JCommander-sdv/src/Utilities/BasicOperation.java

```diff
@@ -47,37 +46,56 @@ public static boolean copyItem(File source, File target){
      * Method to copy recursively a file.
      * @param source Name + source path for the file to be copied
      * @param target Name + target path for the file 
+     * @return True when the file was copied successfully
      */
-    private static void copyFile(File source, File target){
+    private static boolean copyFile(File source, File target) {
+        boolean result;
         try {
             //REPLACE_EXISTING should be editable
             Files.copy(source.toPath(), target.toPath(), REPLACE_EXISTING);
+            result = true;
         } catch (IOException ex) {
             Logger.getLogger(BasicOperation.class.getName()).log(Level.SEVERE, null, ex);
+            result = false;
         }
+        return result;
     }
     
     /**
      * Method to copy recursively a directory with 
      * files and other directories. 
      * @param source Name + source path for the file to be copied
      * @param target Name + target path for the file
+     * @return true when the directory was copied successfully
       */
-    private static void copyDirectory(File source, File target){
+    private static boolean copyDirectory(File source, File target) {
         if (!target.exists()) {
-            target.mkdir();
+            target.mkdirs();
         }
-        for (String file : source.list()) {
-            copyItem(new File(source, file), new File(target, file));
+        boolean result = false;
+        if (source.exists()) {
+            for (String file : source.list()) {
+                File sourceFile = new File(source, file);
+                File targetFile = new File(target, file);
+                copyItem(sourceFile, targetFile);
+            }
         }
+        
+        if (Arrays.equals(source.list(), target.list())) {
```

Comment:
Review if you could implement this "if (Arrays.equals(source.list(), target.list() && target.exists())) {
     result = true;
} ".